Bug in our sparsehash setup

In our sparsehash setup, we have the following:

https://github.com/LLNL/axom/blob/9c6ebdcf475da4d1ba9d574ac26c2c6feec3d2dc/src/thirdparty/CMakeLists.txt#L128-L134

This errors out for ``gcc@8.1+`` when Fortran in not enabled with the following message:

```cmake
CMake Error at cmake/blt/cmake/BLTPrivateMacros.cmake:230 (target_link_libraries):
  Error evaluating generator expression:

    $<COMPILE_LANGUAGE:Fortran>

  $<COMPILE_LANGUAGE:...> Unknown language.
```

--- a/src/thirdparty/CMakeLists.txt
+++ b/src/thirdparty/CMakeLists.txt
@@ -126,11 +126,13 @@ if (AXOM_ENABLE_SPARSEHASH)
                 $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/thirdparty>)
 
     # Disable warning introduced in gcc@8.1 related to how sparsehash casts memory
-    if(C_COMPILER_FAMILY_IS_GNU)
-        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "8.1")
-            blt_add_target_compile_flags(TO sparsehash FLAGS 
-              $<$<NOT:$<COMPILE_LANGUAGE:Fortran>>:-Wno-class-memaccess>)
-        endif()
+    if(C_COMPILER_FAMILY_IS_GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "8.1")
+      if (ENABLE_FORTRAN)
+        blt_add_target_compile_flags(TO sparsehash FLAGS
+            $<$<NOT:$<COMPILE_LANGUAGE:Fortran>>:-Wno-class-memaccess>)
+      else()
+        blt_add_target_compile_flags(TO sparsehash FLAGS -Wno-class-memaccess>)
+      endif()
     endif()
 
     install(TARGETS              sparsehash

